src/SonarScanner.MSBuild.PreProcessor/Roslyn/RoslynRuleSetGenerator.cs

```diff
@@ -148,20 +83,29 @@ private static string GetPartialRepoKey(SonarRule rule, string language)
             }
         }
 
-        private string GetRequiredPropertyValue(string propertyKey)
+        private static bool IsSupportedRuleRepo(IGrouping<string, SonarRule> analyzerRules) =>
+            !string.IsNullOrEmpty(analyzerRules.Key);
+
+        private Rules CreateRulesElement(IGrouping<string, SonarRule> analyzerRules)
         {
-            if (!this.sonarProperties.TryGetValue(propertyKey, out var propertyValue))
+            var partialRepoKey = analyzerRules.Key;
+            return new Rules
             {
-                var message = $"Property does not exist: {propertyKey}. This property should be set by the plugin in SonarQube.";
+                AnalyzerId = GetRequiredPropertyValue($"{partialRepoKey}.analyzerId"),
+                RuleNamespace = GetRequiredPropertyValue($"{partialRepoKey}.ruleNamespace"),
+                RuleList = analyzerRules.Select(CreateRuleElement).ToList()
+            };
+        }
 
-                if (propertyKey.StartsWith(string.Format(SONARANALYZER_PARTIAL_REPO_KEY, "vbnet")))
-                {
-                    message = message + " Possible cause: this Scanner is not compatible with SonarVB 2.X. If necessary, upgrade SonarVB latest in SonarQube.";
-                }
```

Comment:
It's time to let the obsolete warning go (for simplicity). SonarVB doesn't exist for a long time already, 7.9 LTS has v8.6